apps/backend/lib/src/auth.ts

```diff
@@ -0,0 +1,77 @@
+import { Request, Response, NextFunction } from "express";
+import * as jwt from "jsonwebtoken";
+import { Algorithm } from "jsonwebtoken";
+
+interface UserToken {
+  userId: number;
+}
+
+export interface AuthenticatedRequest extends Request {
+  user: UserToken;
+}
+
+const SIGNOPTIONS: jwt.SignOptions = {
+  algorithm: "RS256" as Algorithm,
+  expiresIn: "14 days",
+  issuer: "martletplace:user",
+};
+
+export function authenticate_request(
+  req: AuthenticatedRequest,
+  res: Response,
+  next: NextFunction,
+) {
+  const JWT_PUBLIC_KEY =
+    process.env.JWT_PUBLIC_KEY ||
+    (() => {
+      throw new Error("JWT_PUBLIC_KEY is not set");
+    })();
+
+  // @ts-ignore
+  let decoded: jwt.JwtPayload & UserToken;
+
+  const authCookie = req.cookies["authorization"];
+  try {
+    decoded = jwt.verify(authCookie, JWT_PUBLIC_KEY, { algorithms: ["RS256"] });
+  } catch (error) {
+    console.error(error);
+    res.status(401);
```

Comment:
can we have a json message attached to this with a descriptive error

Reply:
same goes for the other returns

Reply:
Api spec says no response body